Parse/src/test/java/com/parse/SubclassTest.java

```diff
@@ -132,15 +137,23 @@ public void testRegisteringSubclassesUsesMostDescendantSubclass() throws Excepti
       // throw an exception in unit test environment.
       ParseCurrentUserController controller = mock(ParseCurrentUserController.class);
       ParseCorePlugins.getInstance().registerCurrentUserController(controller);
-      assertEquals(ParseUser.class, ParseObject.create("_User").getClass());
+      assertEquals(ParseUser.class, ParseObject.create("_User" ).getClass());
       ParseObject.registerSubclass(MyUser.class);
-      assertEquals(MyUser.class, ParseObject.create("_User").getClass());
+      assertEquals(MyUser.class, ParseObject.create("_User" ).getClass());
       ParseObject.registerSubclass(ParseUser.class);
       assertEquals(MyUser.class, ParseObject.create("_User").getClass());
-      ParseObject.registerSubclass(MyUser2.class);
-      assertEquals(MyUser2.class, ParseObject.create("_User").getClass());
+
+      // This is expected to fail as MyUser2 and MyUser are not directly related.
+      try {
+        ParseObject.registerSubclass(MyUser2.class);
+        fail();
+      } catch (IllegalArgumentException ex) {
+        /* expected */
+      }
```

Comment:
This is the inheritance problem we discussed yesterday, and the exception that existed on all other platforms exception Java.